README.md

```diff
@@ -81,6 +81,15 @@ throughputLimit = 600
 -- Whether the integrity check, meant to spot installation issues, is disabled.
 integrityCheckDisabled = false
 
+-- Whether or not authentication is required
+auth.enabled = false 
+-- Authentication tokens table with client names and their tokens for split tokens. 
+auth.tokens = {
+  -- client => clientName, token => Any token. Advice to use UTF-8 only. Length not limited explicitly 
```

Comment:
There is an `eval` feature controlled in the config that allows for callers to execute arbitrary code, if it isn't too much effort then adding that to the hash and controlling for it would be very useful I think

```lua
{ client = "test", token = "Sometest", eval = true },
{ client = "test", token = "Sometest", eval = false },
```

Reply:
Looked into this. 
I like the idea as well, but from a Request<Body> I don't seem to easily receive the called service/method. 
I definitely want to have another go at it later, so maybe a seperate issue for it would be good. 
Maybe someone else does known.

I didn't really want to check the metadata twice (once in the Interceptor and once in the function `eval` itself. 
Not sure if it's due to the version (that uses the generic `http::request::Request` and not `tonic::Request` which might have some more options. 

So if it's alright with you I'd like to suggest to mark this suggestion as a seperate improvement :)